xbmc/dialogs/GUIDialogKeyboardGeneric.cpp

```diff
@@ -198,7 +198,8 @@ bool CGUIDialogKeyboardGeneric::OnAction(const CAction &action)
   bool handled = true;
   if (actionId == (KEY_VKEY | XBMCVK_BACK))
     Backspace();
-  else if (actionId == ACTION_ENTER || (m_isKeyboardNavigationMode && action.GetID() == ACTION_SELECT_ITEM))
+  else if (actionId == ACTION_ENTER ||
+           ((m_isKeyboardNavigationMode || GetFocusedControlID() == CTL_EDIT) && action.GetID() == ACTION_SELECT_ITEM))
     OnOK();
```

Comment:
This looks much clearer split over multiple lines:

```c++
else if (actionId == ACTION_ENTER)
  OnOK();
else if (actionId == ACTION_SELECT_ITEM &&
    (m_isKeyboardNavigationMode || GetFocusedControlID() == CTL_EDIT))
  OnOK();
```

Reply:
-1 this will introduce dupe code.

Reply:
switched the conditions so that the actionId is checked first, like in your suggestion.

Reply:
Looks much better, thanks